swan-cern/templates/fluentd/fluentd_filters.conf.yaml

```diff
@@ -0,0 +1,131 @@
+      # - metric value
+      <filter {metrics}>
+          @type parser
+          key_name log
+
+          reserve_data true
+          emit_invalid_record_to_error true
+
+          <parse>
+              @type grok
+
+              # keep unmatched logs
+              grok_failure_key grokfailure
+
+              <grok>
+                  # messages printed with "metric: metric_key.metric_context", where metric_context can have dots
+                  pattern user: %{DATA:_metric_user_}, host: %{DATA}, metric: (?<_metric_key_>(\w*))(\.)(?<_metric_context_>(.*)), value: %{GREEDYDATA:_metric_value_}
+              </grok>
+              <grok>
+                  # messages printed with "metric: metric_key"
+                  pattern user: %{DATA:_metric_user_}, host: %{DATA}, metric: (?<_metric_key_>(.*))(?<_metric_context_>(.*)), value: %{GREEDYDATA:_metric_value_}
+              </grok>
+          </parse>
+      </filter>        
+
+      # Store extracted metric information in a new "metrics" field, as part of the JSON
+      # of the log message. Use metric key as first attribute under "metrics".
+      # Format: metrics.metric_key.user|context|value
+      <filter {metrics}>
+          @type record_transformer
+          enable_ruby
+          <record>
+              metrics.${record['_metric_key_']}.user ${record['_metric_user_']}
+              metrics.${record['_metric_key_']}.context ${record['_metric_context_']}
+              metrics.${record['_metric_key_']}.value ${record['_metric_value_']}
+
+              # the below field is needed to run queries that group values by the last segment of the metric context (segments split by '.')
+              # which looks like "metrics.{record['_metric_key_']}_lastelementofmetric_context = {record['metric_value']}"
+              # for example, for a record with below fields
+              #     "data.metrics.spawn.context = LCG_xxx.some_spark_cluster.exception_class"
+              #     "data.metrics.spawn.value = None"
+              # it emits a field "data.metrics.spawn_exception_class = None
+              metrics.${record['_metric_key_']}_${record['_metric_context_'].gsub('-','_').split('.')[-1]} ${record['_metric_value_']}
+          </record>
+          remove_keys ["_metric_user_", "_metric_key_", "_metric_context_", "_metric_value_"]
+      </filter>
+
+      # Emit general log records with type "logs"
+      # Accessible in OpenSearch with pattern monit_private_swan_logs_logs
+      <filter {logs}>
+          @type record_transformer
+          <record>
+              type "logs"
+          </record>
+      </filter>
+
+      # Emit metric log records with type "metrics"
+      # Accessible in OpenSearch with pattern monit_private_swan_logs_metrics
+      <filter {metrics}>
+          @type record_transformer
+          <record>
+              type "metrics"
+          </record>
+      </filter>  
+
+      <filter {logs,metrics}>
```

Comment:
This filter block lost its original comment, see [here](https://github.com/swan-cern/swan-charts/blob/master/swan-cern/values.yaml#L382-L383).

Reply:
I added it back. Initially I had not noticed it had been lost.